### ISSUE TITLE
🐛 [Bug Report] Flurry (War/Sham) Talent doesn't refresh if you land a critical strike at 1 stack

### ISSUE THREAD

**Issue by DramatimeWoW** [cmangos/issues#2085]
### Expected behavior
After critical strike you get 3 stacks of Flurry, one charge is consumed per auto attack as intended,
but when one charge of Flurry is left and you crit - buff just disappears. The buff refreshes correctly if you crit with 2 or 3 stacks.  This seems like an bug that existed prior to patch 2.4 but was fixed.

https://wow.gamepedia.com/Flurry_(warrior_ability)

The Burning Crusade Patch 2.4.0 (2008-03-25): Flurry will properly refresh if a crit occurs with 1 charge left.

### Version & Environment

Client Version: "2.4.3" (TBC)

CMaNGOS Repo & Commit Hash: [TBC](https://github.com/cmangos/mangos-tbc/tree/a440e856392dd55ed41b6d3a9eba80e79492187c)

Database Repo & Commit Hash: [TBC](https://github.com/cmangos/tbc-db/tree/ad1f6281d31b09bb5eea08d36681d296f9de00da)

Operating System: Win 10

### Steps to reproduce
1. Make a shaman or warrior and spec into the flurry talent
2. auto attack something and proc flurry
3. get a critical strike on a single flurry stack
4. observe that it doesn't refresh
5. Profit

### Crashlog
None

**Comment by Yatzii93** [comment]
Confirmed goosed.

It's due to how it executes the code here - 
void Unit::ProcDamageAndSpellFor(ProcSystemArguments& argData, bool isVictim)

Considering the proc is always a success (procs of melee hits to consume charge) the additional proc from a crit isn't being factored in. The result will always be the removal of the aura if the charge is only 1.

A way I fixed it but probz "shit code" :P 

>  case SPELL_AURA_PROC_CANT_TRIGGER:
>                     continue;
>                 case SPELL_AURA_PROC_FAILED:
>                     procSuccess = false;
>                     break;
>                 case SPELL_AURA_PROC_OK:
>                     **if (argData.procExtra && triggeredByHolder)
>                         if(triggeredByHolder->GetSpellProto()->SpellVisual == 2759 && argData.procExtra == 2)
>                             spellprocced = true;**
>                     break;
> 

&& a final check before aura removal 

>  for (RemoveSpellList::const_iterator i = removedSpells.begin(); i != removedSpells.end(); ++i)
>             **if(!spellprocced)**
>                 RemoveAurasDueToSpell(*i);

ofc initilize the spellprocced bool up at the top of the function "bool spellprocced = false;"

Basically the code is looking for the crit proc flag. If pesent don't remove aura.

**Comment by Yatzii93** [comment]
Maybe something that checks for proc flags and cross-references it with the spells proc flags. If the same don't remove.

**Comment by killerwife** [comment]
*flies to the rescue
![image](https://user-images.githubusercontent.com/7995382/72210394-f6f47a00-34ba-11ea-96fe-65d56c3b8f93.png)

**Comment by Yatzii93** [comment]
I'll give this ago, thanks Killer!

**Comment by berserkingyadis** [comment]
This is still happening.
